redesign_tests/test_subgenerators.py

```diff
@@ -1,23 +1,37 @@
 import unittest
 
-from redesign.generator_builder import Generator
-import pandas as pd
 import dataprofiler as dp
+import pandas as pd
 
-class TestSubgenerators(unittest.TestCase):
+from redesign.generator_builder import Generator
 
+
+class TestSubgenerators(unittest.TestCase):
     @classmethod
     def setUpClass(cls):
-        tab_data = dp.Data("redesign_tests/tabular.csv")
-        cls.tab_profile = dp.Profiler(tab_data)
+        data = dp.Data("redesign_tests/iris.csv")
 
-        unstruct_data = pd.Series(['first string', 'second string'])
-        cls.unstruct_profile = dp.Profiler(unstruct_data, profiler_type='unstructured')   
+        profile_options = dp.ProfilerOptions()
+        profile_options.set(
+            {
+                "data_labeler.is_enabled": False,
+                "correlation.is_enabled": True,
+                "multiprocess.is_enabled": False,
+            }
+        )
+        cls.tab_profile = dp.Profiler(
+            data, profiler_type="structured", options=profile_options
+        )
+
+        unstruct_data = pd.Series(["first string", "second string"])
+        cls.unstruct_profile = dp.Profiler(
+            unstruct_data, profiler_type="unstructured", options=profile_options
+        )
 
     def test_synthesize_tab(self):
-        result = Generator(self.tab_profile).synthesize()
-        self.assertEqual(result, "Synthesized tabular data!")
-    
+        result = Generator(profile=self.tab_profile).synthesize(100)
+        self.assertEqual(len(result), 100)
+
     def test_synthesize_unstruct(self):
-        result = Generator(self.unstruct_profile).synthesize()
-        self.assertEqual(result, "Synthesized unstructured data!")
+        with self.assertRaises(NotImplementedError):
+            Generator(profile=self.unstruct_profile).synthesize()
```

Comment:
add a test for `GraphGenerator` `NotImplementedErrror`

Reply:
used the `assertRegex` to validate the error that we specified in the code is actually returned